### ISSUE TITLE
Unused underscore variables not being ignored by lint rules

### ISSUE THREAD

**Issue by chmeyers** [Microsoft/TypeScript#19614]
**TypeScript Version:**  2.7.0-dev.20171031, 2.6.1  (Regression from 2.5.3)

**Expected behavior:**
Variables with names that begin with an underscore do not cause an error if not used.

**Actual behavior:**
components/panel/panel.ts(22,11): error TS6133: '_someVariable' is declared but its value is never read.

**Comment by andy-ms** [comment]
Underscores may be used to indicate unused parameters. A local variable such as `const _ = 0;` will still be flagged. Could you provide a complete example to show a regression?

**Comment by chmeyers** [comment]
Originally noticed this on this variable: 
 https://github.com/yebrahim/datalab/blob/52ab35257d8c338ba10dfbf631fa756e603e37c7/sources/web/datalab/polymer/components/auth-panel/auth-panel.ts#L22

Let me see if I can make a self-contained repro.

**Comment by andy-ms** [comment]
`this._signedIn` is written to but it doesn't appear to be read anywhere.

**Comment by chmeyers** [comment]
Repro:
class TestClass {
  private _shouldBeIgnored: boolean;

  _someFunction() {
    this._shouldBeIgnored = true;
  }
}

tsc --noUnusedLocals test.ts
(Works in 2.5.3, fails on 2.6.1 (and latest nightly))

**Comment by chmeyers** [comment]
In our particular case these variables are read by the html page.

**Comment by andy-ms** [comment]
We don't support underscore ignores for private fields, only for parameters.

**Comment by nikklassen** [comment]
My use case for this is array destructing. I have a function that returns a "tuple" (an array like `[int, string]`) and in some cases I would like to ignore one of the elements i.e. `const [_foo, bar] = func()` but keep the tuple-like semantics. `const bar = func()[1]` no communicates that this is a tuple.

**Comment by aluanhaddad** [comment]
@nikklassen you can use a leading `,` as in
```ts
const [ , bar ] = func();
```

**Comment by petemcwilliams** [comment]
This warning

> is written to but it doesn't appear to be read anywhere

Would be more useful than the one I was getting in version 2.6.1

> is declared but its value is never read.

Because it is a new check I was searching for quite a while to figure this out, seeing the variable was being used (written to) in the ts.
